### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/resources/assets/html/partials/tdp-invest-login.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
+ <div class="col-md-6 col-md-offset-3">
+     <h2>Login</h2>
+    <div ng-show="error" class="alert alert-danger">{{error}}</div>
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/html/partials/tdp-invest-login.html]: According to html & css conventions, the class attribute should always be specified first on a html element. Can you change in all templates?

Comment (mateuszs-kainos) [inline src/main/resources/assets/html/partials/tdp-invest-login.html]: Yes

File: src/main/resources/assets/html/partials/tdp-invest-register.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            <input type="text" name="username" id="username" class="form-control" ng-model="user.username" required />
+            <span ng-show="form.username.$dirty && form.username.$error.required" class="help-block">Username is required</span>
+        </div>
+        <div class="form-group" ng-class="{ 'has-error': form.password.$dirty && form.password.$error.required }">
+            <label for="password">Password</label>
+            <input type="password" name="password" id="password" class="form-control" ng-model="user.password" required />
+            <span ng-show="form.password.$dirty && form.password.$error.required" class="help-block">Password is required</span>
+        </div>
+        <div class="form-actions">
+            <button type="submit" ng-disabled="form.$invalid || dataLoading" class="btn btn-primary">Register</button>
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/html/partials/tdp-invest-register.html]: You don't really need the dataLoading parameter - if you set $scope.promise = {some promise} you can check for a "resolved" property on the promise.

File: src/main/resources/assets/js/application/controllers/tdpInvestLoginController.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+define(['angular', 'application/tdpInvestModule', 'application/services/tdpAuthService'], function(angular, tdpInvestModule) {
+    tdpInvestModule.controller("tdpInvestLoginController", function($scope, $rootScope, $location, $stateParams, tdpAuthService) {
+
+        tdpAuthService.ClearCredentials();
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/controllers/tdpInvestLoginController.js]: By convention - javascript method names start with lower case, e.g. clearCredentials()

File: src/main/resources/assets/js/application/services/tdpAuthService.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+            return function () {
+                return { success: false, message: error };
+            };
+        }
+
+        return service;
+    }]);
+
+    // Base64 encoding service used by AuthenticationService
+    var Base64 = {
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/services/tdpAuthService.js]: I suggest you move this to a separate service

File: src/main/resources/assets/js/application/services/tdpAuthService.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
+define(['angular','application/tdpInvestModule', 'ngCookies'], function(angular, tdpInvestModule) {
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/services/tdpAuthService.js]: We started the naming the filed with projectName prefix - this should be therefore named tdpInvestAuthService

File: src/main/resources/assets/js/application/services/tdpAuthService.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        service.Register = function(username, password) {
+            return $http.post('/api/register',  { mail: username, password: password }).then(handleSuccess, function (response) {
+                if(response.status == 409){
+                    return { success: false, message: "Email address already in use." };
+                }
+                return { success: false, message: "Registration failed." };
+
+            });
+        };
+
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/services/tdpAuthService.js]: Could you clean up the new lines a bit, there's a lot of unnecessary ones

File: src/main/resources/assets/js/application/services/tdpAuthService.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
+define(['angular','application/tdpInvestModule', 'ngCookies'], function(angular, tdpInvestModule) {
+    tdpInvestModule.service('tdpAuthService', ['$cookieStore', '$http', '$rootScope', '$timeout', function($cookieStore, $http, $rootScope) {
+        var service = {};
+
+        service.Login = function(username, password) {
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/services/tdpAuthService.js]: method names should start with a lowercase letter

File: src/main/resources/assets/js/application/tdpInvestModuleConfig.js [hunk 8]
```diff
@@ -1,11 +1,15 @@
...
-    tdpInvestModule.config(function($stateProvider) {
+    , 'application/controllers/tdpInvestLoginController'
+    , 'application/controllers/tdpInvestRegisterController'
+    , 'application/controllers/tdpInvestNavbarController'
+], function (angular, tdpInvestModule) {
+    tdpInvestModule.config(function($stateProvider, $urlRouterProvider) {
         $stateProvider
             .state("tdp", {
                 url: "/tdp",
+                // authenticate: true,
```
Comment (wojciechh-kainos) [inline src/main/resources/assets/js/application/tdpInvestModuleConfig.js]: a leftover comment?

File: src/main/resources/assets/html/partials/tdp-invest-login.html (hunk 1 above)
Comment (mateuszs-kainos) [inline src/main/resources/assets/html/partials/tdp-invest-login.html]: Done

**Pull request merged**